lightning/src/ln/functional_tests.rs

```diff
@@ -6964,7 +6966,7 @@ fn test_bump_penalty_txn_on_revoked_htlcs() {
 	connect_blocks(&nodes[0].block_notifier, 20, 145, true, header_145.bitcoin_hash());
 	{
 		let mut node_txn = nodes[0].tx_broadcaster.txn_broadcasted.lock().unwrap();
-		assert_eq!(node_txn.len(), 1); //TODO: fix check_spend_remote_htlc lack of watch output
+		assert_eq!(node_txn.len(), 0); // Spending of revoked htlc output by claiming transaction remove request as expected
```

Comment:
I'm super confused as to why adding a watch output should ever result in fewer transactions broadcast. This comment doesn't really tell me anything - it follows up on the previous comment, but I'm confused what its trying to say.

Reply:
Okay, I added a comment explaining the reason why bugfix results in fewer transaction broadcast (and reference to issue). Reason is watching output now means when our justice tx confirms onchain and spend revoked htlc output, this is reported to bumping logic which stop dutifully new justice tx generation. Yeah that's not straightforward but I think now that's a right behavior.